src/Compilers/VisualBasic/Portable/Parser/ParseExpression.vb

```diff
@@ -1393,66 +1393,74 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
         ' Lines: 16425 - 16425
         ' .Parser::ParseArguments( [ _In_ ParseTree::ArgumentList** Target ] [ _Inout_ bool& ErrorInConstruct ] )
 
-        Private Function ParseArguments(ByRef unexpected As GreenNode, Optional RedimOrNewParent As Boolean = False) As CodeAnalysis.Syntax.InternalSyntax.SeparatedSyntaxList(Of ArgumentSyntax)
+        Private Function ParseArguments(ByRef unexpected As GreenNode, Optional RedimOrNewParent As Boolean = False, Optional attributeListParent As Boolean = False) As CodeAnalysis.Syntax.InternalSyntax.SeparatedSyntaxList(Of ArgumentSyntax)
             Dim arguments = _pool.AllocateSeparated(Of ArgumentSyntax)()
 
-            Do
-                ' Check for the first keyword argument.
+            Dim allowNonTrailingNamedArguments = _scanner.Options.LanguageVersion.AllowNonTrailingNamedArguments()
+            Dim seenNames As Boolean = False
 
+            Do
+                Dim isNamed As Boolean = False
                 If (CurrentToken.Kind = SyntaxKind.IdentifierToken OrElse CurrentToken.IsKeyword()) AndAlso
                     PeekToken(1).Kind = SyntaxKind.ColonEqualsToken Then
 
-                    ParseNamedArguments(arguments)
-                    Exit Do
-
+                    seenNames = True
+                    isNamed = True
                 End If
 
                 Dim comma As PunctuationSyntax = Nothing
+                If isNamed Then
+                    Dim argumentName As IdentifierNameSyntax = ParseIdentifierNameAllowingKeyword()
 
-                Select Case CurrentToken.Kind
-                    Case SyntaxKind.CommaToken
-                        TryGetTokenAndEatNewLine(SyntaxKind.CommaToken, comma)
-                        arguments.Add(InternalSyntaxFactory.OmittedArgument())
-                        arguments.AddSeparator(comma)
-                        Continue Do
-
-                    Case SyntaxKind.CloseParenToken
-                        If arguments.Count > 0 Then
-                            arguments.Add(InternalSyntaxFactory.OmittedArgument())
-                        End If
-                        Exit Do
-
-                    Case Else
-                        Dim argument = ParseArgument(RedimOrNewParent)
+                    Dim colonEquals As PunctuationSyntax = Nothing
+                    TryGetTokenAndEatNewLine(SyntaxKind.ColonEqualsToken, colonEquals)
+                    Dim namedArgument = SyntaxFactory.SimpleArgument(SyntaxFactory.NameColonEquals(argumentName, colonEquals), ParseExpressionCore())
+                    arguments.Add(namedArgument)
+
+                ElseIf CurrentToken.Kind = SyntaxKind.CommaToken Then
+                    TryGetTokenAndEatNewLine(SyntaxKind.CommaToken, comma)
+                    Dim argument = ReportNonTrailingNamedArgumentIfNeeded(InternalSyntaxFactory.OmittedArgument(), seenNames, allowNonTrailingNamedArguments)
```

Comment:
>InternalSyntaxFactory.OmittedArgument() [](start = 74, length = 39)

It feels like we shouldn't just create an OmmitedArgument after a named argument when attributeListParent is true. We probably should try to parse it as named.